src/main/java/com/kentyou/featurelauncher/cli/FeatureLauncherCli.java

```diff
@@ -0,0 +1,428 @@
+			+ "Launcher provides an executable JAR file which allows a Feature "
+			+ "to be launched from the command line.";
+	static final String VERSION = HEADING + " 1.0";
+	static final int EXITCODE_SUCCESS = 0;
+	static final String REMOTE_ARTIFACT_REPOSITORY_URI_VALUE = "REMOTE_ARTIFACT_REPOSITORY_URI";
+
+	@ArgGroup(exclusive = true, multiplicity = "1", order = -10)
+	private FeatureFile featureFile;
+
+	@Option(names = { "-a",
+			"--artifact-repository" }, paramLabel = "uri [key=value]", description = "Specifies an artifact repository URI and optionally one "
+					+ "or more configuration properties for that artifact repository, "
+					+ "such as those described in Remote Repositories on page 1381. "
+					+ "This property may be repeated to add more than one artifact "
+					+ "repository.", order = -9, mapFallbackValue = REMOTE_ARTIFACT_REPOSITORY_URI_VALUE)
+	private Map<Object, Object> remoteArtifactRepositories;
+
+	@Option(names = { "-d", "--decorator" }, paramLabel = "class name", description = "Provides "
+			+ "the name of a decorator class that should be used when launching "
+			+ "the feature. The decorator class must be public, available on the "
+			+ "classpath, and have a public zero-argument constructor. This "
+			+ "property may be repeated to add more than one decorator.", order = -8)
+	private List<Class<?>> decorators;
+
+	@Option(names = { "-e",
+			"--extension-handler" }, paramLabel = "extension name=class name", description = "Provides the name of an extension, and the extension handler "
+					+ "class that should be used to handle the extension when launching "
+					+ "the feature. The extension handler class must be public, available "
+					+ "on the classpath, and have a public zero-argument constructor. This "
+					+ "property may be repeated to add more than one extension handler.", order = -7)
+	private Map<String, Class<?>> extensionHandlers;
+
+	@Option(names = { "-l",
+			"--launch-property" }, paramLabel = "key=value", description = "Provides one or more launch properties that should be passed "
+					+ "to the framework when it is launched.", order = -6)
+	private Map<String, String> frameworkProperties;
+
+	@Option(names = { "-v",
+			"--variable-override" }, paramLabel = "key=value", description = "Provides one or more variables that should be used to set or "
+					+ "override variables defined in the feature.", order = -5)
+	private Map<String, Object> variables;
+
+	@Option(names = { "-c",
+			"--configuration" }, paramLabel = "key=value", description = "Provides one or more configuration properties that should "
+					+ "be used to control implementation specific behaviour.", order = -4)
+	private Map<String, Object> configuration;
+
+	@Option(names = {
+			"--dry-run" }, description = "Evaluates all options, processes them and displays output, but does not launch framework. Hidden option used for testing", hidden = true)
```

Comment:
This must be `--impl-dry-run` according to the specification so that it doesn't clash with future command options.